cli.go

```diff
@@ -0,0 +1,387 @@
+	}
+	emitCLIResponse(emptyResponse)
+}
+
+func callRPCListPeer(rpcAddr string) {
+	conn, err := grpc.Dial(rpcAddr, grpc.WithInsecure())
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to connect to RPC server at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	defer conn.Close()
+	client := pbrpc.NewPocClient(conn)
+	listPeerReq := &pbrpc.RPCListPeerRequest{}
+	logger.Debugf("rpcclient:ListPeer: sending=%v", listPeerReq)
+	res, err := client.ListPeer(context.Background(), listPeerReq)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to call RPC ListPeer at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	peers := &PeerResponse{
+		Peers: res.Peers.Peers,
+	}
+	emitCLIResponse(peers)
+}
+
+func callRPCListTopicPeer(rpcAddr string, topics []string) {
+	conn, err := grpc.Dial(rpcAddr, grpc.WithInsecure())
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to connect to RPC server at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	defer conn.Close()
+	client := pbrpc.NewPocClient(conn)
+	listTopicPeerReq := &pbrpc.RPCListTopicPeerRequest{
+		Topics: topics,
+	}
+	logger.Debugf("rpcclient:ListTopicPeer: sending=%v", listTopicPeerReq)
+	res, err := client.ListTopicPeer(context.Background(), listTopicPeerReq)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to call RPC ListTopicPeer at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	topicPeers := make(map[string][]string)
+	for topic, peers := range res.TopicPeers {
+		topicPeers[topic] = peers.Peers
+	}
+	emitCLIResponse(topicPeers)
+}
+
+func callRPCRemovePeer(rpcAddr string, peerID peer.ID) {
+	conn, err := grpc.Dial(rpcAddr, grpc.WithInsecure())
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to connect to RPC server at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	defer conn.Close()
+	client := pbrpc.NewPocClient(conn)
+	removePeerReq := &pbrpc.RPCRemovePeerRequest{
+		PeerID: peerIDToString(peerID),
+	}
+	logger.Debugf("rpcclient:removePeerReq: sending=%v", removePeerReq)
+	_, err = client.RemovePeer(context.Background(), removePeerReq)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to call RPC listpeer at %v, err: %v", rpcAddr, err)
+		emitCLIFailure(errMsg)
+	}
+	emitCLIResponse(emptyResponse)
+}
+
+type CLIResponse struct {
+	Status bool
+	Data   interface{}
+}
+
+type CLIFailure struct {
+	Status  bool
+	Message string
+}
+
+type ShardResponse struct {
+	ShardIDs []ShardIDType
+}
+
+type PeerResponse struct {
+	Peers []string
+}
+
+type TopicPeerResponse struct {
+	TopicPeers map[string]([]string)
+}
+
+var emptyResponse interface{}
+
+func marshalToJSONString(data interface{}) string {
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to marshal object: %v", err)
+		emitCLIFailure(errMsg)
+	}
+	dataStr := string(dataBytes)
+	return dataStr
+}
+
+func emitCLIFailure(msg string) {
+	resp := &CLIFailure{
+		Status:  false,
+		Message: msg,
+	}
+	respStr := marshalToJSONString(resp)
```

Comment:
Will this results in a `marshalToJSONString` -> `emitCLIFailure` -> `marshalToJSONString` loop?

Reply:
Yes, you're right. It will cause a recursion. Will fix it.